src/Compilers/CSharp/Portable/Lowering/SpillSequenceSpiller.cs

```diff
@@ -274,7 +270,7 @@ private BoundStatement UpdateStatement(BoundSpillSequenceBuilder builder, BoundS
                 builder.AddStatement(statement);
             }
 
-            var result = _F.Block(builder.GetLocals(), builder.GetStatements());
+            var result = new BoundBlock(statement.Syntax, builder.GetLocals(), builder.GetStatements()) { WasCompilerGenerated = true };
```

Comment:
If this `BoundBlock` represents a closure span, the `Syntax` field will be used to calculate `ClosureDebugInfo.SyntaxOffset`, but the `Syntax` field was being set to the `Syntax` of the current `BoundSequencePoint` rather than `Syntax` of the original block.